File: JeongHanUl/week_5/boj_1806.java
package nestnet_algorithm_2025_1_advanced.JeongHanUl.week_5;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class boj_1806 {
    //public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int target = Integer.parseInt(st.nextToken());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) arr[i] = Integer.parseInt(st.nextToken());

        int start = 0, end = 0;
        int sum = arr[0], length = Integer.MAX_VALUE;
        while (start <= end) {
            if (sum < target) {
                end++;
                if (end >= n) break;
                sum += arr[end];
            } else {
                length = Math.min(length, end - start + 1);
                sum -= arr[start];
                start++;
            }
        }

        if (length == Integer.MAX_VALUE) System.out.println(0);
        else System.out.println(length);
    }
}

File: ChaeJangWoo/week5/boj_1806.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

int main()
{
    int N, S;
    cin >> N >> S;
    vector<int> arr(N);
    for (int i = 0; i < N; ++i)
    {
        cin >> arr[i];
    }

    int minLength = INT_MAX;
    int start = 0, end = 0, sum = 0;

    while (true)
    {
        if (sum >= S)
        {
            minLength = min(minLength, end - start);
            sum -= arr[start++];
        }
        else if (end == N)
        {
            break;
        }
        else
        {
            sum += arr[end++];
        }
    }

    if (minLength == INT_MAX)
        cout << 0 << endl;
    else
        cout << minLength << endl;

    return 0;
}
